app/components/Authentication/AwsEc2/AwsEc2.jsx

```diff
@@ -1,28 +1,704 @@
+
+    ec2ConfigSchema = {
+        access_key: '',
+        endpoint: undefined,
+        secret_key: ''
+    };
+
+    roleConfigSchema = {
+        bound_ami_id: undefined,
+        bound_account_id: undefined,
+        bound_region: undefined,
+        bound_vpc_id: undefined,
+        bound_subnet_id: undefined,
+        bound_iam_role_arn: undefined,
+        bound_iam_instance_profile_arn: undefined,
+        role_tag: undefined,
+        ttl: undefined,
+        max_ttl: undefined,
+        period: undefined,
+        policies: [],
+        allow_instance_migration: undefined,
+        disallow_reauthentication: false
+    }
+
     constructor(props) {
         super(props);
         this.state = {
-
+            baseUrl: `/auth/aws-ec2/${this.props.params.namespace}/`,
+            baseVaultPath: `auth/${this.props.params.namespace}`,
+            ec2Roles: [],
+            configObj: this.ec2ConfigSchema,
+            newConfigObj: this.ec2ConfigSchema,
+            newRoleConfig: this.roleConfigSchema,
+            selectedRoleId: '',
+            newRoleId: '',
+            newSecretBtnDisabled: false,
+            openNewRoleDialog: false,
+            openEditRoleDialog: false,
+            deleteUserPath: '',
+            selectedTab: "roles",
+            isBackendConfigured: false
         };
 
+        _.bindAll(
+            this,
+            'listEc2Roles',
+            'getEc2AuthConfig',
+            'createUpdateConfig',
+            'createUpdateRole'
+        );
+
+    }
+
+    listEc2Roles() {
+        tokenHasCapabilities(['list'], `${this.state.baseVaultPath}/role`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/role`, { list: true }, null)
+                    .then((resp) => {
+                        let roles = resp.data.data.keys;
+                        this.setState({ ec2Roles: _.valuesIn(roles) });
+                    })
+                    .catch((error) => {
+                        if (error.response.status !== 404) {
+                            snackBarMessage(error);
+                        } else {
+                            this.setState({ ec2Roles: [] });
+                        }
+                    });
+            })
+            .catch(() => {
+                snackBarMessage(new Error("Access denied"));
+            })
     }
+
+    getEc2AuthConfig() {
+        tokenHasCapabilities(['read'], `${this.state.baseVaultPath}/config/client`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/config/client`, null, null)
+                    .then((resp) => {
+                        let config = resp.data.data;
+                        this.setState({
+                            configObj: update(this.state.configObj,
+                                {
+                                    access_key: { $set: (config.access_key ? config.access_key : null) },
```

Comment:
why set it to null? if it is undefined use it and it will falsey most likey where you make the check later